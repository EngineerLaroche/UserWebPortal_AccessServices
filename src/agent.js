import superagentPromise from 'superagent-promise';
import _superagent from 'superagent';

const superagent = superagentPromise(_superagent, global.Promise);
const USER_API = 'http://localhost:4000';

//const encode = encodeURIComponent;
const responseBody = res => res.body;

let token = null;
const tokenPlugin = req => {
  if (token) {
    req.set('authorization', `Token ${token}`);
  }
}

const requests = {
  del: url =>
    superagent.del(`${USER_API}${url}`).use(tokenPlugin).then(responseBody),
  get: url =>
    superagent.get(`${USER_API}${url}`).use(tokenPlugin).then(responseBody),
  put: (url, body) =>
    superagent.put(`${USER_API}${url}`, body).use(tokenPlugin).then(responseBody),
  post: (url, body) =>
    superagent.post(`${USER_API}${url}`, body).use(tokenPlugin).set('Content-Type', 'application/json').then(responseBody)
};

export const Auth = {
  current: () =>
    requests.get('/user'),
  login: (email, password) =>
    requests.post('/user/login', { user: { email, password } }),
  register: (firstname, lastname, email, password, role) =>
    requests.post('/user/create', { user: { firstname, lastname, email, password, role } }),
  logout: () =>
    requests.post('/user/logout', {}),
  save: user =>
    requests.put('/user/update', { user }),
  deactivate: (email) =>
    requests.put('/user/deactivate', { email }),
  reactivate: (email) =>
    requests.put('/user/reactivate', { email })
};

const Referent = {
  register: (firstname, lastname, title, workphone, cellphone, fax, email, preferences) =>
    requests.post('/referent/create', { referent: { firstname, lastname, title, workphone, cellphone, fax, email, preferences } }),
  associate: (organismeId, email) =>
    requests.post('/referent/organismereferent', { association: { organismeId, email } }),
  all: () =>
    requests.post('/referents', {}),
  save: referent =>
    requests.put('/referent/update', { referent }),
  clear: (email) =>
    requests.del('/referent/clearpreferences', { association: email }),
  delete: id =>
    requests.del('/referent/delete', { id }),
  preference: (preferenceId, email) =>
    requests.post('/referent/preferencereferent', { association: { preferenceId, email } })
};

const OrganismeReferent = {
  register: (name, email, address, phone, fax, website, state, organisme) =>
    requests.post('/organismereferent/create', { organismeReferent: { name, email, address, phone, fax, website, state, organisme } }),
  all: () =>
    requests.post('/organismereferents', {}),
  save: organismeReferent =>
    requests.put('/organismereferent/update', { organismeReferent }),
  delete: id =>
    requests.del('/organismereferent/delete', { id })
};

const Organisme = {
  register: (name, email, address, phone, fax) =>
    requests.post('/organisme/create', { organisme: { name, email, address, phone, fax } }),
  all: () =>
    requests.post('/organismes', {}),
  save: organisme =>
    requests.put('/organisme/update', { organisme }),
  delete: id =>
    requests.del('/organisme/delete', { id })
};

const PointService = {
  register: (name, email, phone, fax, nocivique, street, city, province, postalcode) =>
    requests.post('/pointservice/create', { pointService: { name, email, phone, fax, nocivique, street, city, province, postalcode} }),
  all: () =>
    requests.post('/pointservices', {}),
  save: pointService =>
    requests.put('/pointservice/update', { pointService }),
  delete: id =>
    requests.del('/pointservice/delete', { id })
};

const Service = {
  register: (name, description, tarifParent, tarifCISSS, state, stateSubvention, datePrice) =>
    requests.post('/service/create', { service: { name, description, tarifParent, tarifCISSS, state, stateSubvention, datePrice} }),
  all: () =>
    requests.post('/services', {}),
  save: service =>
    requests.put('/service/update', { service }),
  delete: id =>
    requests.del('/service/delete', { id })
};

const DemandeService = {
  register: (serviceType, noDossierFamille, date, frequence, assumeFrais, motifs, transport, parent, secondParent, enfants, dispoFirstParent, dispoSecondParent) =>
    requests.post('/demandeservice/create', { demandeservice: { serviceType, noDossierFamille, date, frequence, assumeFrais, motifs, transport, parent, secondParent, enfants, dispoFirstParent, dispoSecondParent} }),
  all: () =>
    requests.post('/demandeservices', {}),
  save: (id, date, frequence, assumeFrais, motifs, transport, parent, secondParent, enfants, dispoFirstParent, dispoSecondParent) =>
    requests.put('/demandeservice/update', { demandeservice: { id, date, frequence, assumeFrais, motifs, transport, parent, secondParent, enfants, dispoFirstParent, dispoSecondParent} }),
  delete: id =>
    requests.del('/demandeservice/delete', { id }),
  addMotif: (motif) =>
    requests.post('/motifs/add', {newMotif: motif})
};

const Users = {
  all: () =>
    requests.get('/users')
};

export default {
  Auth,
  Users,
  Referent,
  OrganismeReferent,
  Organisme,
  PointService,
  Service,
  DemandeService,
  setToken: _token => { token = _token; }
};
