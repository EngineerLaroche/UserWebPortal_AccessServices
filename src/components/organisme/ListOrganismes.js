import { Link } from "react-router-dom";
import Request from 'superagent';
import React from 'react';
import { connect } from 'react-redux';
import { FormattedMessage } from 'react-intl';
import { GETORGANISMES, ORGANISMESEARCH } from '../../constants/actionTypes';

const mapStateToProps = state => ({
    organismes: '',
    search: ''
});

const mapDispatchToProps = dispatch => ({
    onLoad: () =>
        dispatch({ type: GETORGANISMES }),
    onSearch: () =>
        dispatch({ type: ORGANISMESEARCH, payload: this.search })
});
export class ListOrganismes extends React.Component {
    constructor(){
        super();
        this.state = {
            search: '',
            organismes: []
        }
    }

    componentWillMount() {
        var url = "http://localhost:4000/organismes"
        Request.get(url).then((response) => {
            console.log(response.body);
            var data = response.body.rows;
            this.setState({ 'organismes': data });
        });

    }

    render() {
        const data = this.state.organismes;

        return (

            <div className="container page">

                <hr />
                <h1 className="text-xs-center">
                    <FormattedMessage
                        id="listorganismes.title"
                        defaultMessage="Liste des Organismes" />
                </h1>
                <hr />
                <br />
                <br />

                <div className="row">
                    <div className="col-md-8 offset-md-3 col-xs-12">
                        <div className="text-xs-center">

                            <label>
                                <FormattedMessage
                                    id="listorganismes.search"
                                    defaultMessage="Rechercher par nom :" />
                            </label>
                            <br />
                            <input type="text" value={this.search}>
                            </input>
                            <button onClick={this.props.onSearch}>Go</button>
                        </div>
                        <br />
                        <br />

                        <div className="text-xs-left">
                            {(data != null) && Object.values(data).map((d) =>
                                <li key={d.Email}>
                                    <Link to={"/organisme/" + d.Email} className="nav-link">
                                        {d.Name}
                                    </Link>
                                    <hr />
                                </li>
                            )}
                        </div>
                    </div>
                </div>
            </div>
        );
    }
} export default connect(mapStateToProps, mapDispatchToProps)(ListOrganismes);