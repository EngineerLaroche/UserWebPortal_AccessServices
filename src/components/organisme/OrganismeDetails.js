import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { FormattedMessage } from 'react-intl';
import Request from 'superagent';

const EditOrganismeDetails = props => {
  if (props.isOrganisme) {
    return (
      <Link to={`/organisme/edit/${props.currentOrganisme.email}`} className="btn btn-primary">
        <i className="ion-gear-a"> </i>
        <FormattedMessage
          id="userdetails.modify"
          defaultMessage="Modifier" />
      </Link>
    );
  }
};

const mapStateToProps = state => ({
  currentOrganisme: state.common.currentOrganisme,
  organisme: state.common.organisme
});

const mapDispatchToProps = dispatch => ({});

export class OrganismeDetails extends React.Component {

  constructor() {
    super();
    this.state = {
      currentOrganisme: ''
    }
  }

  componentWillMount() {
    var url = "http://localhost:4000/organisme/" + this.props.match.params.email;
    Request.get(url).then((response) => {
      console.log(response.body);
      var data = response.body.organisme;
      this.setState({ 'organisme': data });
      this.setState({ 'currentOrganisme': data });
    });
  }

  render() {
    const currentOrganisme = this.state.currentOrganisme;
    if (!currentOrganisme) {
      return null;
    }

    return (

      <div className="profile-page">
        <div className="container page">

          <hr />
          <h1 className="text-xs-center">
            <FormattedMessage
              id="organismedetails.title"
              defaultMessage="Profil de l'organisme" />
          </h1>
          <hr />
          <br />
          <br />
          <div className="container">
            <div className="row">
              <div className="text-xs-left">

                <h4>
                  <FormattedMessage
                    id="createorganisme.name"
                    defaultMessage="Nom" />
                  :  {currentOrganisme.name}
                </h4>
                <h4>
                  <FormattedMessage
                    id="createorganisme.email"
                    defaultMessage="Email" />
                  :  {currentOrganisme.email}
                </h4>
                <br />
                <h4>
                  <FormattedMessage
                    id="createorganisme.tel"
                    defaultMessage="Numéro de téléphone" />
                  :  {currentOrganisme.phone}
                </h4>
                <h4>
                  <FormattedMessage
                    id="createorganisme.fax"
                    defaultMessage="Numéro de fax" />
                  :  {currentOrganisme.fax}
                </h4>
                <h4>
                  <FormattedMessage
                    id="createorganisme.address"
                    defaultMessage="Adresse" />
                  :  {currentOrganisme.address.nocivique} {currentOrganisme.address.street}, {currentOrganisme.address.city}, {currentOrganisme.address.province}
                </h4>
                <h4>
                  <FormattedMessage
                    id="createorganisme.postalcode"
                    defaultMessage="Code postal" />
                  :  {currentOrganisme.address.postalcode}
                </h4>
                <br />

                <EditOrganismeDetails isOrganisme={true}
                  currentOrganisme={currentOrganisme} />

              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(OrganismeDetails);
