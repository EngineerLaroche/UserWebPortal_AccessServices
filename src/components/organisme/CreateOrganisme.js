import ListErrors from '../ListErrors';
import React from 'react';
import agent from '../../agent';
import { FormattedMessage, intlShape, injectIntl } from 'react-intl';
import { connect } from 'react-redux';

import {
    UPDATE_FIELD_AUTH,
    CREATE_ORGANISME,
    REGISTER_PAGE_UNLOADED
} from '../../constants/actionTypes';

const mapStateToProps = state => ({ ...state.auth });

const mapDispatchToProps = dispatch => ({
    onChangeName: value =>
        dispatch({ type: UPDATE_FIELD_AUTH, key: 'name', value }),
    onChangePhone: value =>
        dispatch({ type: UPDATE_FIELD_AUTH, key: 'phone', value }),
    onChangeNocivique: value =>
        dispatch({ type: UPDATE_FIELD_AUTH, key: 'nocivique', value }),
    onChangeFax: value =>
        dispatch({ type: UPDATE_FIELD_AUTH, key: 'fax', value }),
    onChangeEmail: value =>
        dispatch({ type: UPDATE_FIELD_AUTH, key: 'email', value }),
    onChangeStreet: value =>
        dispatch({ type: UPDATE_FIELD_AUTH, key: 'street', value }),
    onChangeCity: value =>
        dispatch({ type: UPDATE_FIELD_AUTH, key: 'city', value }),
    onChangeProvince: value =>
        dispatch({ type: UPDATE_FIELD_AUTH, key: 'province', value }),
    onChangePostalcode: value =>
        dispatch({ type: UPDATE_FIELD_AUTH, key: 'postalcode', value }),
    onSubmit: (name, email, address, phone, fax) => {
        const payload = agent.Organisme.register(name, email, address, phone, fax);
        dispatch({ type: CREATE_ORGANISME, payload })
    },
    onUnload: () =>
        dispatch({ type: REGISTER_PAGE_UNLOADED })
});

export class CreateOrganisme extends React.Component {
    constructor() {
        super();
        this.changeName = ev => this.props.onChangeName(ev.target.value);
        this.changePhone = ev => this.props.onChangePhone(ev.target.value);
        this.changeState = ev => this.props.onChangeState(ev.target.value);
        this.changeNocivique = ev => this.props.onChangeNocivique(ev.target.value);
        this.changeFax = ev => this.props.onChangeFax(ev.target.value);
        this.changeEmail = ev => this.props.onChangeEmail(ev.target.value);
        this.changeCity = ev => this.props.onChangeCity(ev.target.value);
        this.changeStreet = ev => this.props.onChangeStreet(ev.target.value);
        this.changeProvince = ev => this.props.onChangeProvince(ev.target.value);
        this.changePostalcode = ev => this.props.onChangePostalcode(ev.target.value);
        this.submitForm = (name, email, nocivique, street, city, province, postalcode, phone, fax) =>
            ev => {
                ev.preventDefault();
                var address = { nocivique, street, city, province, postalcode };
                this.props.onSubmit(name, email, address, phone, fax);
            }
    }

    componentWillUnmount() {
        this.props.onUnload();
    }

    render() {
        const intl = this.props.intl;
        const name = this.props.name;
        const phone = this.props.phone;
        const fax = this.props.fax;
        const email = this.props.email;
        const nocivique = this.props.novicique;
        const street = this.props.street;
        const city = this.props.city;
        const province = this.props.province;
        const postalcode = this.props.postalcode;

        return (
            <div className="auth-page">
                <div className="container page">

                    <hr />
                    <h1 className="text-xs-center">
                        <FormattedMessage
                            id="createorganisme.title"
                            defaultMessage="Créer Organisme" />
                    </h1>
                    <hr />
                    <br />
                    <br />

                    <ListErrors errors={this.props.errors} />

                    <div className="row">
                        <div className="col-md-8 offset-md-3 col-xs-12">

                            <form onSubmit={this.submitForm(name, email, nocivique, street, city, province, postalcode, phone, fax).bind(this)}>
                                <h3 className="text-xs-center">Informations</h3>
                                <fieldset>
                                    <fieldset className="form-group">
                                        <input
                                            className="form-control form-control-lg"
                                            type="text"
                                            placeholder={intl.formatMessage({ id: "createorganisme.name", defaultMessage: "Nom" })}
                                            value={this.props.name}
                                            onChange={this.changeName} />
                                    </fieldset>
                                </fieldset>
                                <fieldset>
                                    <fieldset className="form-group">
                                        <input
                                            className="form-control form-control-lg"
                                            type="text"
                                            placeholder={intl.formatMessage({ id: "createorganisme.email", defaultMessage: "Email" })}
                                            value={this.props.email}
                                            onChange={this.changeEmail} />
                                    </fieldset>
                                </fieldset>
                                <fieldset>
                                    <fieldset className="form-group">
                                        <input
                                            className="form-control form-control-lg"
                                            type="text"
                                            placeholder={intl.formatMessage({ id: "createorganisme.tel", defaultMessage: "Numéro de téléphone" })}
                                            value={this.props.phone}
                                            onChange={this.changePhone} />
                                    </fieldset>
                                </fieldset>
                                <fieldset>
                                    <fieldset className="form-group">
                                        <input
                                            className="form-control form-control-lg"
                                            type="text"
                                            placeholder={intl.formatMessage({ id: "createorganisme.fax", defaultMessage: "Numéro de fax" })}
                                            value={this.props.fax}
                                            onChange={this.changeFax} />
                                    </fieldset>
                                </fieldset>
                                <h3 className="text-xs-center">
                                    <FormattedMessage
                                        id="createorganisme.address"
                                        defaultMessage="Adresse" />
                                </h3>
                                <fieldset>
                                    <fieldset className="form-group">
                                        <input
                                            className="form-control form-control-lg"
                                            type="text"
                                            placeholder={intl.formatMessage({ id: "createorganisme.civnum", defaultMessage: "Numéro civique" })}
                                            value={this.props.nocivique}
                                            onChange={this.changeNocivique} />
                                    </fieldset>
                                </fieldset>
                                <fieldset>
                                    <fieldset className="form-group">
                                        <input
                                            className="form-control form-control-lg"
                                            type="text"
                                            placeholder={intl.formatMessage({ id: "createorganisme.street", defaultMessage: "Rue" })}
                                            value={this.props.street}
                                            onChange={this.changeStreet} />
                                    </fieldset>
                                </fieldset>
                                <fieldset>
                                    <fieldset className="form-group">
                                        <input
                                            className="form-control form-control-lg"
                                            type="text"
                                            placeholder={intl.formatMessage({ id: "createorganisme.city", defaultMessage: "Ville" })}
                                            value={this.props.city}
                                            onChange={this.changeCity} />
                                    </fieldset>
                                </fieldset>
                                <fieldset>
                                    <fieldset className="form-group">
                                        <input
                                            className="form-control form-control-lg"
                                            type="text"
                                            placeholder={intl.formatMessage({ id: "createorganisme.province", defaultMessage: "Province" })}
                                            value={this.props.province}
                                            onChange={this.changeProvince} />
                                    </fieldset>
                                </fieldset>
                                <fieldset>
                                    <fieldset className="form-group">
                                        <input
                                            className="form-control form-control-lg"
                                            type="text"
                                            placeholder={intl.formatMessage({ id: "createorganisme.postalcode", defaultMessage: "Code postal" })}
                                            value={this.props.postalcode}
                                            onChange={this.changePostalcode} />
                                    </fieldset>
                                </fieldset>
                                <br />
                                <div className="text-xs-center">
                                    <button
                                        className="btn btn-lg btn-primary btn-block"
                                        type="submit"
                                        disabled={this.props.inProgress}>
                                        <FormattedMessage
                                            id="header.create"
                                            defaultMessage="Créer" />
                                    </button>
                                </div>
                            </form>
                        </div>
                    </div>
                </div>
            </div>
        );
    }
}

CreateOrganisme.propTypes = {
    intl: intlShape.isRequired
};
CreateOrganisme = injectIntl(CreateOrganisme);

export default connect(mapStateToProps, mapDispatchToProps)(CreateOrganisme);
