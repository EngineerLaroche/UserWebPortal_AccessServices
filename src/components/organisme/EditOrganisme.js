import ListErrors from '../ListErrors';
import React from 'react';
import agent from '../../agent';
import Request from 'superagent';
import { FormattedMessage, intlShape, injectIntl } from 'react-intl';
import { connect } from 'react-redux';
import {
    SETTINGS_SAVED,
    SETTINGS_PAGE_UNLOADED,
    UPDATE_ORGANISME
} from '../../constants/actionTypes';

export class EditOrganismeForm extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            id: props.currentOrganisme.id,
            name: props.currentOrganisme.name,
            phone: props.currentOrganisme.phone,
            addressId: props.addressId,
            nocivique: props.currentOrganisme.address.nocivique,
            street: props.currentOrganisme.address.street,
            city: props.currentOrganisme.address.city,
            province: props.currentOrganisme.address.province,
            postalcode: props.currentOrganisme.address.postalcode,
            fax: props.currentOrganisme.fax,
            email: props.currentOrganisme.email
        };

        this.updateState = field => ev => {
            const state = this.state;
            const newState = Object.assign({}, state, { [field]: ev.target.value });
            this.setState(newState);
        };

        this.submitForm = ev => {
            ev.preventDefault();

            const currentOrganisme = Object.assign({}, this.state);
            this.props.onSubmitForm(currentOrganisme);
        };
    }

    componentWillReceiveProps(nextProps) {
        if (nextProps.currentOrganisme) {
            this.setState(Object.assign({}, this.state, {
                id: nextProps.currentOrganisme.id,
                addressId: nextProps.addressId,
                name: nextProps.currentOrganisme.name,
                phone: nextProps.currentOrganisme.phone,
                street: nextProps.currentOrganisme.address.street,
                city: nextProps.currentOrganisme.address.city,
                nocivique: nextProps.currentOrganisme.address.nocivique,
                province: nextProps.currentOrganisme.address.province,
                postalcode: nextProps.currentOrganisme.address.postalcode,
                fax: nextProps.currentOrganisme.fax,
                email: nextProps.currentOrganisme.email
            }));
        }
    }

    render() {
        const intl = this.props.intl;
        return (
            <form onSubmit={this.submitForm}>
                <fieldset>
                    <h3 className="text-xs-center">
                        Informations
                    </h3>
                    <fieldset className="form-group">
                        <input
                            className="form-control form-control-lg"
                            type="text"
                            placeholder={this.props.name}
                            value={this.state.name}
                            onChange={this.updateState('name')} />
                    </fieldset>

                    <fieldset className="form-group">
                        <input
                            className="form-control form-control-lg"
                            type="email"
                            placeholder={this.props.email}
                            value={this.state.email}
                            onChange={this.updateState('email')} />
                    </fieldset>

                    <fieldset className="form-group">
                        <input
                            className="form-control form-control-lg"
                            type="phone"
                            placeholder={this.props.fax}
                            value={this.state.fax}
                            onChange={this.updateState('fax')} />
                    </fieldset>

                    <fieldset className="form-group">
                        <input
                            className="form-control form-control-lg"
                            type="text"
                            placeholder={this.props.tel}
                            value={this.state.phone}
                            onChange={this.updateState('phone')} />
                    </fieldset>

                    <h3 className="text-xs-center">
                        <FormattedMessage
                            id="createorganisme.address"
                            defaultMessage="Adresse" />
                    </h3>
                    <fieldset className="form-group">
                        <input
                            className="form-control form-control-lg"
                            type="text"
                            placeholder={this.props.numciv}
                            value={this.state.nocivique}
                            onChange={this.updateState('nocivique')} />
                    </fieldset>

                    <fieldset className="form-group">
                        <input
                            className="form-control form-control-lg"
                            type="text"
                            placeholder={this.props.street}
                            value={this.state.street}
                            onChange={this.updateState('street')} />
                    </fieldset>

                    <fieldset className="form-group">
                        <input
                            className="form-control form-control-lg"
                            type="text"
                            placeholder={this.props.city}
                            value={this.state.city}
                            onChange={this.updateState('city')} />
                    </fieldset>

                    <fieldset className="form-group">
                        <input
                            className="form-control form-control-lg"
                            type="text"
                            placeholder={this.props.province}
                            value={this.state.province}
                            onChange={this.updateState('province')} />
                    </fieldset>

                    <fieldset className="form-group">
                        <input
                            className="form-control form-control-lg"
                            type="text"
                            placeholder={this.props.postalcode}
                            value={this.state.postalcode}
                            onChange={this.updateState('postalcode')} />
                    </fieldset>
                    <h3 className="text-xs-center">
                        <FormattedMessage
                            id="editorganisme.active.text"
                            defaultMessage="Statut d'activité" />
                    </h3>
                    <fieldset className="form-group">
                        <select
                            className="form-control form-control-xl"
                            value={this.state.state}
                            onChange={this.updateState('state')}>
                            <option key=" ">{this.props.activetext}</option>
                            <option value="0">
                                {this.props.active}
                            </option>
                            <option value="1">
                                {this.props.inactive}
                            </option>
                        </select>
                    </fieldset>
                    <br />
                    <button
                        className="btn btn-lg btn-primary btn-block"
                        type="submit"
                        disabled={this.state.inProgress}>
                        <FormattedMessage
                            id="settings.update"
                            defaultMessage="Mettre à jour" />
                    </button>
                </fieldset>
            </form>
        );
    }
}

const mapStateToProps = state => ({
    ...state.settings,
    currentOrganisme: state.currentOrganisme
});

const mapDispatchToProps = dispatch => ({
    onSubmitForm: organisme =>
        dispatch({ type: UPDATE_ORGANISME, payload: agent.Organisme.save(organisme) }),
    onUnload: () => dispatch({ type: SETTINGS_PAGE_UNLOADED })
});

export class EditOrganisme extends React.Component {
    constructor() {
        super();
        this.state = {
            currentOrganisme: ''
        }
    }
    componentWillMount() {
        if (this.props.currentOrganisme) {
            Object.assign(this.state, {
                name: this.props.currentOrganisme.name,
                phone: this.props.currentOrganisme.phone,
                state: this.props.currentOrganisme.state,
                address: this.props.currentOrganisme.address,
                fax: this.props.currentOrganisme.fax,
                email: this.props.currentOrganisme.email
            });
        } else {
            var url = "http://localhost:4000/organisme/" + this.props.match.params.email;
            Request.get(url).then((response) => {
                console.log(response.body);
                var data = response.body.organisme;
                this.setState({ 'currentOrganisme': data });
                this.setState({'addressId': data.address});
            });
        }
    }
    render() {
        const intl = this.props.intl;
        const currentOrganisme = this.state.currentOrganisme;
        const addressId = this.state.addressId;
        if (!currentOrganisme) {
            return null;
        }
        return (
            <div className="settings-page">
                <div className="container page">

                    <hr />
                    <h1 className="text-xs-center">
                        <FormattedMessage
                            id="organismedetails.title"
                            defaultMessage="Paramètres de l'organisme" />
                    </h1>
                    <hr />
                    <br />
                    <br />

                    <div className="row">
                        <div className="col-md-8 offset-md-2 col-xs-12">

                            <ListErrors errors={this.props.errors}></ListErrors>
                            <EditOrganismeForm
                                currentOrganisme={currentOrganisme}
                                name={intl.formatMessage({
                                    id: "createorganisme.name",
                                    defaultMessage: "Nom"
                                })}
                                email={intl.formatMessage({
                                    id: "createorganisme.email",
                                    defaultMessage: "Email"
                                })}
                                fax={intl.formatMessage({
                                    id: "createorganisme.fax",
                                    defaultMessage: "Numéro de fax"
                                })}
                                tel={intl.formatMessage({
                                    id: "createorganisme.tel",
                                    defaultMessage: "Numéro de téléphone"
                                })}
                                numciv={intl.formatMessage({
                                    id: "createorganisme.civnum",
                                    defaultMessage: "Numéro civique"
                                })}
                                street={intl.formatMessage({
                                    id: "createorganisme.street",
                                    defaultMessage: "Rue"
                                })}
                                city={intl.formatMessage({
                                    id: "createorganisme.city",
                                    defaultMessage: "Ville"
                                })}
                                province={intl.formatMessage({
                                    id: "createorganisme.province",
                                    defaultMessage: "Province"
                                })}
                                postalcode={intl.formatMessage({
                                    id: "createorganisme.postalcode",
                                    defaultMessage: "Code postal"
                                })}
                                activetext={intl.formatMessage({
                                    id: "editorganisme.active.text",
                                    defaultMessage: "Activité de l'organisme"
                                })}
                                active={intl.formatMessage({
                                    id: "editorganisme.active",
                                    defaultMessage: "Actif"
                                })}
                                inactive={intl.formatMessage({
                                    id: "editorganisme.inactive",
                                    defaultMessage: "Inactif"
                                })}
                            
                                addressId={addressId}
                                onSubmitForm={this.props.onSubmitForm} />
                        </div>
                    </div>
                </div>
            </div>
        );
    }
}


EditOrganisme.propTypes = {
    intl: intlShape.isRequired
};
EditOrganisme = injectIntl(EditOrganisme);

export default connect(mapStateToProps, mapDispatchToProps)(EditOrganisme);
