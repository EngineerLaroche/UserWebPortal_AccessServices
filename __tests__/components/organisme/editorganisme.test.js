import React from 'react';
import {mount, shallow} from 'enzyme';
import {EditOrganisme, EditOrganismeForm} from 'components/organisme/EditOrganisme';
import agent from 'agent';

describe('EditOrganisme is rendering', () => {
    let component;
    let form;
    let currentOrganisme = {
        address: 1,
        email: "test@organisme.com",
        fax: "454789692231",
        id: 1,
        name: "Test Inc.",
        phone: "1-800-888-1111",
    };
    let onSubmitForm= jest.fn();


    beforeAll(() => {
        agent.Auth.login('yross@gucci.com', '12345');
        agent.Auth.current();
        form = shallow(<EditOrganismeForm
            currentOrganisme={currentOrganisme}
            onSubmitForm={onSubmitForm}/>);
        component = shallow(<EditOrganisme/>);
    });

    afterAll(() => {
        agent.Auth.logout();
    });

    it('EditOrganisme renders components', () => {
        expect(component).toMatchSnapshot();
    });
});
