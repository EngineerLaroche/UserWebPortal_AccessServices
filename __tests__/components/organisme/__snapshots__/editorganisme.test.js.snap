// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditOrganisme is rendering EditOrganisme renders components 1`] = `undefined`;
